smoke_test/tests/sam_smoke_test_case.py

```diff
@@ -0,0 +1,23 @@
+from functools import cache
+from unittest import TestCase
+from urllib.parse import urljoin
+
+import requests
+from requests import Response
+
+
+class SamSmokeTestCase(TestCase):
+    SAM_HOST = None
+    USER_TOKEN = None
+
+    @staticmethod
+    def build_sam_url(path: str) -> str:
+        assert SamSmokeTestCase.SAM_HOST, "ERROR - SamSmokeTests.SAM_HOST not properly set"
+        return urljoin(f"https://{SamSmokeTestCase.SAM_HOST}", path)
```

Comment:
tested this out with command:
```
▶ python smoke_test.py https://sam.dsde-dev.broadinstitute.org
```

and got this unhelpful error:
```
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='https', port=443): Max retries exceeded with url: /version (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x103557160>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
```

turns out it is doubling up on `https://`.

I think this could be personal preference, but since I'm usually going to copy the URL from the browser (which automatically copies the scheme), I think it would be helpful to remove the `https://` prefix here

```suggestion
        return urljoin(f"{SamSmokeTestCase.SAM_HOST}", path)
```
(tested the change locally and it works)

another reason to make this change is that the error message is much better if a user leaves out the scheme prefix:
```
requests.exceptions.MissingSchema: Invalid URL '/version': No scheme supplied. Perhaps you meant http:///version?
```

(Also, running against localhost might use http instead of https but I'm not 100% sure about that.)

Reply:
fixed to optionally allow the protocol to be specified